microsoft/community-organization-operations-suite
# Save information while in Offline Mode

Word done with @karlolsonuc 

**What** 
 - Resolve #534 
 - Being able to Add/Edit element while in _Offline Mode_.

**Why**
 - Save as much as possible locally that is already existing before toggling `offline` mode.

**How**
 - Leverage _Apollo Client_ tools and library to cache, [queue](https://github.com/unchartedsoftware/apollo-link-queue) requests.
 - Setup a _Service Worker_to cache has many UI assets as possible, while ignoring tracking and GraphQL requests

**Testing**
 - change the config file to turn `Service Workers` to `true`.
 - Open a page containing information editable or to add.
 - Turn _Offline Mode_ on.
 - Add or Edit an item.
 - The data should not change.
 - Turn _Offline Mode_ off.
 - The data should update with the new information submitted while in _Offline Mode_.

**Anything else**
 - More tweaking needs to be improved, like deleting all the data while _Offline Mode_ is turned on.
 - We can also improves the process by using _Apollo Optimistic answers_ to Mutations.
